DataAccessLayer/Decorators/DataLoggerService.cs

```diff
@@ -0,0 +1,64 @@
+﻿using SocialNetwork.DataContext;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DataAccessLayer.Decorators
+{
+   public class DataLoggerService<T> : IDataService<T> where T: class,new()
+   {
+      private IDataService<T> entity;
+      private string path = "DataServiceLog.txt";
+      
+
+      public DataLoggerService(IDataService<T> entity)
+      {
+         this.entity = entity;
+      }
+
+      public void AddEntity(T entity)
+      {
+         string logString = string.Format("{0}({1})", nameof(this.AddEntity), typeof(T).Name);
```

Comment:
I hinted at the lecture that you should use reflection (MethodBase.GetCurrentMethod) to get method name)